src/App.jsx

```diff
@@ -1,46 +1,96 @@
 import 'bulma/css/bulma.css';
+import cn from 'classnames/bind';
+import { useState } from 'react';
 import './App.scss';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  { id: 1, name: 'Dumplings' },
+  { id: 2, name: 'Carrot' },
+  { id: 3, name: 'Eggs' },
+  { id: 4, name: 'Ice cream' },
+  { id: 5, name: 'Apple' },
+  { id: 6, name: 'Bread' },
+  { id: 7, name: 'Fish' },
+  { id: 8, name: 'Honey' },
+  { id: 9, name: 'Jam' },
+  { id: 10, name: 'Garlic' },
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ABC = 'abc';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getReorderedGoods(goods, sortType, isReversed) {
+  const visibleGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortType === SORT_BY_LENGTH) {
+    visibleGoods.sort((good1, good2) => good1.name.length - good2.name.length);
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortType === SORT_BY_ABC) {
+    visibleGoods.sort((good1, good2) => good1.name.localeCompare(good2.name));
+  }
```

Comment:
You are interacting with the DOM directly by using the `sort` method. This is a violation of the first item in the checklist. Try to use React's built-in features to manipulate the DOM as much as possible.